### ISSUE TITLE
Losses out of order in RNNLearner.get_preds(..., with_loss=True, ordered=True)

### ISSUE THREAD

**Issue by dcato98** [fastai/fastai#1678]
**Describe the bug**
In `fastai/text/learner.py`, When calling `get_preds()` on an `RNNLearner` with `with_loss=True` and `ordered=True`, the losses aren't ordered correctly.

**To reproduce**
```
from fastai.text import *

# setup learner with IMDB data
path = untar_data(URLs.IMDB_SAMPLE)
data_clas = TextClasDataBunch.from_csv(path, 'texts.csv', bs=32)
learn = text_classifier_learner(data_clas, AWD_LSTM, drop_mult=0.5)

# get preds
probs, y_trues, reported_loss = learn.get_preds(with_loss=True, ordered=True)

# manually calculate the loss
rng = 0  # either number or a range, e.g. range(0,3)
calc_loss = F.cross_entropy(np.log(probs[rng]).reshape(-1,len(learn.data.classes)), # log(probs) reverses the softmax (ignoring the original scale)
                y_trues[rng].reshape(-1), reduction='none')

# check whether losses match
assert np.all(np.array(calc_loss) - np.array(reported_loss[rng]) < 0.0001)
```

**Expected behavior**
Losses are in the same order as the probs and y_trues. Assertion passes.

**Suggested fix**
In `fastai/text/learner.py`, change this line in `RNNLearner.get_preds()`:
`preds[0],preds[1] = preds[0][reverse_sampler],preds[1][reverse_sampler]`
To this:
`for el in preds: el = el[reverse_sampler]`

And maybe add a test along the lines of the code to reproduce (above), replacing the hard-coded  loss function with the learner's loss function (I am not familiar enough with the library to run the loss in a generalized way).

**Installation details**
```text
=== Software === 
python        : 3.7.2
fastai        : 1.0.46.dev0
fastprogress  : 0.1.19
torch         : 1.0.1
torch cuda    : 9.0 / is available
torch cudnn   : 7005 / is enabled

=== Hardware === 
torch devices : 2
  - gpu0      : GeForce GTX 1080 Ti
  - gpu1      : GeForce GTX 1080 Ti

=== Environment === 
platform      : Windows-10-10.0.17763-SP0
conda env     : fastai-dev
no nvidia-smi is found
```

**Comment by sgugger** [comment]
Oh, I had forgotten that `get_preds` could return three elements sometimes, good catch.

**Comment by dcato98** [comment]
Wow that was quick! Thank you.
